### DISCUSSION TITLE
What is the best way to use this in spring app

### DISCUSSION THREAD

**Discussion by pradeepkusingh** [vladimir-bukhtoyarov/bucket4j#4] [Q&A] [answered]

**Comment by vladimir-bukhtoyarov** [comment]
There are no specific aspects related for Spring.
- If you do not plan to share bucket between multiple beans then just create bucket inside bean and use it. 
- If you are going to share bucket between multiple beans then create bucket via implementation of FactoryBean or method annotated by @Bean annotation.
- If you plan to share bucket between multiple application on same JEE container(like tomcat) or just want to bucket should still alive after application redeployment then create bucket via JNDI in way specific for concrete JEE container.

Is this answer enaugh for you or you asked for something else?

**Comment by pradeepkusingh** [comment]
Thanks, I made a mistake I was re initializing bucket for each call. Now I
added in constructor of bean.

On Thursday, February 18, 2016, Vladimir Bukhtoyarov <
notifications@github.com> wrote:

> There are no specific aspects related for Spring.
> - If you do not plan to share bucket between multiple beans then just
>   create bucket inside bean and use it.
> - If you are going to share bucket between multiple beans then create
>   bucket via implementation of FactoryBean or method annotated by @Bean
>   https://github.com/Bean annotation.
> - If you plan to share bucket between multiple application on same JEE
>   container(like tomcat) or just want to bucket should still alive after
>   application redeployment then create bucket via JNDI in way specific for
>   concrete JEE container.
> 
> Is it ok for you, or you are interesting in something else?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/vladimir-bukhtoyarov/bucket4j/issues/4#issuecomment-186065497
> .

## 

Thanks,
Pradeep Singh
Please excuse typos.

**Comment by MarcGiffing** [comment, marked as answer]
If someone is searching for a Spring integration:

I've started with a Spring Boot Starter project for bucket4j.
[bucket4j-spring-boot-starter](https://github.com/MarcGiffing/bucket4j-spring-boot-starter)
It configures bucket4j automatically if a CacheManager (JCache) is found (configured with spring-boot-starter-cache)

**Comment by vladimir-bukhtoyarov** [comment]
@MarcGiffing 
looks very nice. Please notify when done, I will mention your library in [this list](https://github.com/vladimir-bukhtoyarov/bucket4j#third-party-libraries-which-uses-bucket4j)
